apps/dashboard/pkg/migration/migrate_test.go

```diff
@@ -66,11 +84,75 @@ func TestMigrate(t *testing.T) {
 	}
 }
 
-func testMigration(t *testing.T, dash map[string]interface{}, inputFileName string, targetVersion int) {
+func TestMigrateSingleVersion(t *testing.T) {
+	files, err := os.ReadDir(INPUT_DIR)
+	require.NoError(t, err)
+
+	// Use the same datasource provider as the frontend test to ensure consistency
+	migration.Initialize(testutil.GetTestDataSourceProvider(), testutil.GetTestPanelProvider())
+
+	for _, f := range files {
+		if f.IsDir() {
+			continue
+		}
+
+		// Validate filename format
+		if !strings.HasPrefix(f.Name(), "v") || !strings.HasSuffix(f.Name(), ".json") {
+			t.Fatalf("input filename must use v{N}.{name}.json format, got: %s", f.Name())
+		}
+
+		// Extract version from filename (e.g., v16.grid_layout_upgrade.json -> 16)
+		// This represents the tested version that the file should be migrated TO
+		versionStr := strings.TrimPrefix(f.Name(), "v")
+		dotIndex := strings.Index(versionStr, ".")
+		if dotIndex == -1 {
+			t.Fatalf("input filename must use v{N}.{name}.json format, got: %s", f.Name())
+		}
```

Comment:
This is incorrect, the migration shoult be to v17 in this example case.

Reply:
Ah, actually it is, because the chema version in the `vX` file is `v(X-1)`. This is confusing :D